lib/elastic_apm/opentracing.rb

```diff
@@ -0,0 +1,242 @@
+# frozen_string_literal: true
+
+require 'opentracing'
+
+module ElasticAPM
+  module OpenTracing
+    # @api private
+    class Span
+      def initialize(elastic_span)
+        @elastic_span = elastic_span
+      end
+
+      attr_reader :elastic_span
+
+      def context
+        elastic_span.trace_context
+      end
+
+      def finish(end_time: Time.now)
+        elastic_span.done end_time: Util.micros(end_time)
+
+        case elastic_span
+        when ElasticAPM::Transaction
+          ElasticAPM.end_transaction
+        when ElasticAPM::Span
+          ElasticAPM.end_span
+        end
+      end
+    end
+
+    # @api private
+    class SpanContext
+      def initialize(trace_context)
+        @trace_context = trace_context
+      end
+
+      attr_accessor :trace_context
+    end
+
+    # @api private
+    class Scope
+      def initialize(span, scope_stack, finish_on_close:)
+        @span = span
+        @scope_stack = scope_stack
+        @finish_on_close = finish_on_close
+      end
+
+      attr_reader :span
+
+      def elastic_span
+        span.elastic_span
+      end
+
+      def close
+        @span.finish if @finish_on_close
+        @scope_stack.pop
+      end
+    end
+
+    # @api private
+    class ScopeStack
+      KEY = :__elastic_apm_ot_scope_stack
+
+      def push(scope)
+        scopes << scope
+      end
+
+      def pop
+        scopes.pop
+      end
+
+      def last
+        scopes.last
+      end
+
+      private
+
+      def scopes
+        Thread.current[KEY] ||= []
+      end
+    end
+
+    # @api private
+    class ScopeManager
+      def initialize
+        @scope_stack = ScopeStack.new
+      end
+
+      def activate(span, finish_on_close: true)
+        return active_scope if active_scope && active_scope.span == span
+        scope = Scope.new(span, @scope_stack, finish_on_close: finish_on_close)
+        @scope_stack.push scope
+        scope
+      end
+
+      def active_scope
```

Comment:
This should be called `active` instead of `active_scope`, according to https://github.com/opentracing/opentracing-ruby/blob/master/lib/opentracing/scope_manager.rb#L31